src/main/java/com/activestate/cloudfoundryjenkins/CloudFoundryPushPublisher.java

```diff
@@ -333,38 +333,69 @@ private boolean processOneApp(CloudFoundryClient client, DeploymentInfo deployme
 
     private boolean createApplicationIfNeeded(CloudFoundryClient client, BuildListener listener,
                                               DeploymentInfo deploymentInfo, String appURI) {
-        // Check if app already exists
-        List<CloudApplication> existingApps = client.getApplications();
-        boolean createNewApp = true;
-        for (CloudApplication app : existingApps) {
-            if (app.getName().equals(deploymentInfo.getAppName())) {
-                if (resetIfExists) {
-                    listener.getLogger().println("App already exists, resetting.");
-                    client.deleteApplication(deploymentInfo.getAppName());
-                    listener.getLogger().println("App deleted.");
-                } else {
-                    createNewApp = false;
-                    listener.getLogger().println("App already exists, skipping creation.");
-                }
-                break;
-            }
+        // Get app instance
+        CloudApplication cloudApplication = client.getApplication(deploymentInfo.getAppName());
+        String appTempName = deploymentInfo.getAppName();
+        String appTempURI = appURI;
+
+        // Check if app exists
+        boolean appExists = cloudApplication == null ? false : true;
+
+        if (appExists) {
+           switch (cutoverMethod.toLowerCase()) {
+              case "redeploy":  
+                 listener.getLogger().println("App already exists, redeploying.");
+                 client.deleteApplication(deploymentInfo.getAppName());
+                 listener.getLogger().println("App deleted.");
+              break;
+              case "reroute":  
+                 // Alter app name and set temporary route to new app (blue/green)
+                 listener.getLogger().println("App already exists, redeploying blue/green");
+                 boolean blue = appName.endsWith("-blue") ? true : false;
+                 boolean green = appName.endsWith("-green") ? true : false;
+                 appTempName += blue ? "-green" : green ? "-blue" : "-green" : "-green";
+                 appTempURI += blue ? "-green" : green ? "-blue" : "-green" : "-green";
```

Comment:
Seems like there's one colon too many on those lines, so this doesn't compile.
Also, it's probably better to use plain old ifs there, nesting ternary operators makes things less readable.